### PR TITLE
Feature/search api/request contract, notification, rejected

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

This PR includes the following proposed change(s):

- FAMS3-2486

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested locally and unit tests

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/SearchAgency/AgencyRequestService_Update_Test.cs [hunk 1]
```diff
@@ -1128,7 +1130,7 @@ public async Task wrong_fileId_searchRequestOrdered_ProcessUpdateSearchRequest_s
         }
 
         [Test]
-        public async Task wrong_agencyCode_searchRequestOrdered_ProcessUpdateSearchRequest_should_return_null()
+        public async Task wrong_agencyCode_searchRequestOrdered_ProcessUpdateSearchRequest_should_return_throwException()
```
Comment (rantifamilusi) [inline app/DynamicsAdapter/DynamicsAdapter.Web.Test/SearchAgency/AgencyRequestService_Update_Test.cs:1133]: check warnings

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfile_SearchRequest_Test.cs [hunk 2]
```diff
@@ -49,7 +49,7 @@ public void normal_SearchRequestOrdered_should_map_to_SearchRequestEntity_correc
                         RequestDate = new DateTimeOffset(2018, 9, 28, 0, 0, 0, new TimeSpan(1, 0, 0)),
                         Email = "agencyemail@test.com",
                         InformationRequested = new List<InformationRequested> { InformationRequested.Location, InformationRequested.PHN, InformationRequested.DL },
-                        LocationAddress = "NORTHERN AND INTERIOR CLIENT OFFICE, KAMLOOPS, BC"
+                        LocationCode = "NORTHERN AND INTERIOR CLIENT OFFICE, KAMLOOPS, BC"
```
Comment (rantifamilusi) [inline app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfile_SearchRequest_Test.cs:52]: Did we not introduce code for location s?

File: app/DynamicsAdapter/DynamicsAdapter.Web/SearchAgency/AgencyRequestController.cs [hunk 3]
```diff
@@ -57,7 +57,7 @@ public async Task<IActionResult> CreateSearchRequest(string requestId, [FromBody
                 catch (Exception ex)
                 {
                     _logger.LogError(ex.Message);
-                    return StatusCode(StatusCodes.Status500InternalServerError, new { Message = ex.Message });
+                    return StatusCode(StatusCodes.Status500InternalServerError, new { ReasonCode = ex.Message, Message = ex.InnerException?.Message });
```
Comment (rantifamilusi) [inline app/DynamicsAdapter/DynamicsAdapter.Web/SearchAgency/AgencyRequestController.cs:60]: are we sending this to FMEP ? I don't think we are just checking.

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/SearchAgency/AgencyRequestService_Update_Test.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/SearchAgency/AgencyRequestService_Update_Test.cs:1133]: sorry.

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfile_SearchRequest_Test.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfile_SearchRequest_Test.cs:52]: before, fmep send the whole location address, later, Dragan changed it to code (v, k etc.). So, at first, we have locationAddress, later, we add LocationCode.

File: app/DynamicsAdapter/DynamicsAdapter.Web/SearchAgency/AgencyRequestController.cs (hunk 3 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/SearchAgency/AgencyRequestController.cs:60]: yes, we need to send this to Rejected Message. They ask for specific reason, and reason code. You can see this in Dragan's email on Set.24.

Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web/SearchAgency/AgencyRequestController.cs:60]: I was meaning the exception we don't want to send exception message

**Review by rantifamilusi — APPROVED** [review]

**Pull request merged**